sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -656,6 +656,38 @@ def test_partial_dependence_pipeline():
     )
 
 
+@pytest.mark.parametrize(
+    "features, grid_resolution, n_vals_expected",
+    [
+        (["a"], 10, 10),
+        (["a"], 2, 2),
+    ],
+)
+def test_partial_dependence_binary_model_grid_resolution(
+    features, grid_resolution, n_vals_expected
+):
```

Comment:
I think that the parametrization here is not necssary. We can directly define the feature `"a"` in the function.
And for the grid-resolution, let's focus the test only on the regression that is the case `2`.

Could you add a small docstring to the test to describe the corner case that we test.